src/components/JobsPanelDataInputs/jobsPanelDataInputs.util.js

```diff
@@ -138,3 +166,114 @@ export const handleDelete = (
     )
   })
 }
+
+export const comboboxSelectList = [
+  {
+    className: 'path-type-store',
+    label: 'store',
+    id: 'store://'
+  },
+  {
+    className: 'path-type-s3',
+    label: 'URL',
+    id: 's3://'
+  }
+]
+
+export const S3_INPUT_PATH_TYPE = 's3://'
+
+export const handleInputPathTypeChange = (
+  inputsDispatch,
+  newInput,
+  pathType,
+  pathPlaceholder,
+  newInputDefaultPathProject,
+  currentProject
+) => {
+  if (
+    newInputDefaultPathProject.length > 0 &&
+    pathType === S3_INPUT_PATH_TYPE
+  ) {
+    inputsDispatch({
+      type: inputsActions.SET_NEW_INPUT_DEFAULT_PATH_PROJECT,
+      payload: ''
+    })
+  } else if (newInputDefaultPathProject.length === 0) {
+    inputsDispatch({
+      type: inputsActions.SET_NEW_INPUT_DEFAULT_PATH_PROJECT,
+      payload: `${currentProject}/`
+    })
+  }
+
+  inputsDispatch({
+    type: inputsActions.SET_PATH_PLACEHOLDER,
+    payload:
+      pathType === S3_INPUT_PATH_TYPE && pathPlaceholder.length === 0
+        ? 'bucket/path'
+        : ''
+  })
+  inputsDispatch({
+    type: inputsActions.SET_NEW_INPUT_URL_PATH,
+    payload: ''
+  })
+  inputsDispatch({
+    type: inputsActions.SET_NEW_INPUT_PATH,
+    payload: {
+      pathType: pathType,
+      project: '',
+      artifact: ''
+    }
+  })
+}
+
+export const handleInputPathChange = (inputsDispatch, inputsState, path) => {
+  if (inputsState.newInput.path.pathType === S3_INPUT_PATH_TYPE) {
+    return inputsDispatch({
+      type: inputsActions.SET_NEW_INPUT_URL_PATH,
+      payload: path
+    })
+  }
+
+  const pathItems = path.split('/')
+  const artifactIsEntered = inputsState.artifacts.find(
+    artifact => artifact.id === pathItems[1]
+  )
+
+  if (path.length === 0 && inputsState.newInputDefaultPathProject.length > 0) {
+    inputsDispatch({
+      type: inputsActions.SET_NEW_INPUT_DEFAULT_PATH_PROJECT,
+      payload: ''
+    })
+  }
+
+  if (isNil(pathItems[1]) && inputsState.artifacts.length > 0) {
+    inputsDispatch({
+      type: inputsActions.SET_ARTIFACTS,
+      payload: []
+    })
+  }
+
+  if (
+    pathItems[0] !== inputsState.newInput.path.project ||
+    (!isNil(pathItems[1]) &&
+      pathItems[1] !== inputsState.newInput.path.artifact)
+  ) {
+    inputsDispatch({
+      type: inputsActions.SET_NEW_INPUT_PATH,
+      payload: {
+        ...inputsState.newInput.path,
+        project: pathItems[0],
+        artifact: pathItems[1] ?? inputsState.newInput.path.artifact
+      }
+    })
+  }
+
+  inputsDispatch({
+    type: inputsActions.SET_NEW_INPUT_PROJECT_PATH_ENTERED,
+    payload: pathItems[1] === '' || pathItems[1]?.length > 0
```

Comment:
I'm having trouble understanding the intention of these conditions.

The condition passes when `pathItems[1]` is the empty string `''`, a non-empty string/array, or a function with at least one argument.
The condition fails when `pathItems[1]` is “nullish” (`null` or `undefined`), a boolean, a number, an empty array `[]`, a function with no arguments, or any other object.

See this:

```js
const func = value => value === '' || value?.length > 0

// true
func('')
func('non-empty-string')
func(['non-empty-array'])
func(some => 'function with at least 1 argument')

// false
func(null)
func(undefined)
func(true)
func(false)
func(2)
func([])
func({})
func({ nonEmpty: 'object' })
func(() => {})
```

I assume that since it's a path, it should be a string. If I delete all examples which are not strings, I'm left with:
```js
// true
func('')
func('non-empty-string')

// false
func(null)
func(undefined)
```

So, is the purpose of these conditions just to check whether `pathItem[1]` is not null/undefined?
If so, we can do:
```js
typeof pathItems[1] === 'string'
```